extension/runtime/src/main/java/org/iris_events/runtime/BasicPropertiesProvider.java

```diff
@@ -75,10 +80,17 @@ private AMQP.BasicProperties createAmqpBasicProperties(final String serviceId) {
         final var correlationId = correlationIdProvider.getCorrelationId();
         log.debug("Creating new AMQP.BasicProperties with correlationId: {}", correlationId);
         final var builder = new AMQP.BasicProperties().builder()
-                .correlationId(correlationId)
-                .headers(Map.of());
+                .correlationId(correlationId);
+        final var headers = new HashMap<String, Object>();
         // only set origin_service_id once and never break it as source cannot change
-        Optional.ofNullable(serviceId).ifPresent(id -> builder.headers(Map.of(ORIGIN_SERVICE_ID, serviceId)));
+        Optional.ofNullable(serviceId).ifPresent(id -> headers.put(ORIGIN_SERVICE_ID, serviceId));
+        if (requestContext != null) {
+            // when thread was started by HTTP request we check for x-request-id header and add it to the AMQP message headers
+            Optional.ofNullable(requestContext.getHeaderString(REQUEST_ID))
+                    .ifPresent(httpRequestId -> headers.put(REQUEST_ID, httpRequestId));
```

Comment:
I don't think there is need for introducing new header here, if anything you could set traceid and/or correlation id which are already part of the whole request tracking system.